# Tarea Clase 5

## Tarea 1. Bases de nacimientos 2020–2022

nacimientos_2020.csv nacimientos_2021.csv nacimientos_2022.csv, contienen información de nacimientos por estado.
Realiza lo siguiente:
1. Lee cada archivo usando pd.read_csv()
2. Agrega a cada base una columna llamada "Año", (Ej. nac_2020["Año"] = 2020).
3. Une las tres bases en un solo DataFrame llamado df_nacimientos (Utiliza pd.concat()).
4. Revisa las primeras filas con .head()
5. Revisa cuántas filas y columnas tiene la base final.
6. Calcula el total de nacimientos por año.
7. Identifica qué año tuvo más nacimientos y cuál tuvo menos.

Responde
- ¿Cuántos nacimientos hubo en total en 2020, 2021 y 2022?
- ¿En qué año se registró el mayor número de nacimientos?
- ¿En qué año se registró el menor número de nacimientos?
- ¿La cantidad de nacimientos parece aumentar o disminuir con el tiempo?

In [21]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [22]:
ruta2020="https://raw.githubusercontent.com/jamc88/min-curso/refs/heads/main/Datos/nacimientos_2020.csv"
ruta2021="https://raw.githubusercontent.com/jamc88/min-curso/refs/heads/main/Datos/nacimientos_2021.csv"
ruta2022="https://raw.githubusercontent.com/jamc88/min-curso/refs/heads/main/Datos/nacimientos_2022.csv"

archivos=[(ruta2020,2020),(ruta2021,2021),(ruta2022,2022)]  # se accede usando archivos[i][j]
bases=[]
for ruta,año in archivos:
    df=pd.read_csv(ruta)
    df["Año"]=año
    bases.append(df)
#print(bases)  es un una lista de dataframes
df_nacimientos=pd.concat(bases,ignore_index=True)

df_nacimientos


,Estado,Abreviatura,Regiones,Mujeres,Hombres,No_esp,Total,Año
0,Aguascalientes,AG,Noreste,9966,10404,0,20370,2020
1,Baja California,BC,Noroeste,23539,24406,0,47945,2020
2,Baja California Sur,BS,Noroeste,4982,5099,0,10081,2020
3,Campeche,CM,Sureste,5275,5454,0,10729,2020
4,Coahuila de Zaragoza,CO,Noreste,22594,23221,0,45815,2020
...,...,...,...,...,...,...,...,...
91,Tamaulipas,TM,Noreste,23177,24110,0,47287,2022
92,Tlaxcala,TL,Centro-Sur,9653,9964,1,19618,2022
93,Veracruz,VE,Sureste,53782,55329,0,109111,2022
94,Yucatan,YU,Sureste,13829,14103,0,27932,2022


In [23]:
df_nacimientos.head()

,Estado,Abreviatura,Regiones,Mujeres,Hombres,No_esp,Total,Año
0,Aguascalientes,AG,Noreste,9966,10404,0,20370,2020
1,Baja California,BC,Noroeste,23539,24406,0,47945,2020
2,Baja California Sur,BS,Noroeste,4982,5099,0,10081,2020
3,Campeche,CM,Sureste,5275,5454,0,10729,2020
4,Coahuila de Zaragoza,CO,Noreste,22594,23221,0,45815,2020


In [24]:
df_nacimientos.shape

(96, 8)

In [34]:
nacim=df_nacimientos.set_index("Año")

for i in range(2020,2023):
    nacimientos=nacim.loc[i]["Total"].sum()
    print("Los nacimientos de ",i," son: ",nacimientos)

# de forma alternativa dada como serie
nacimientos=df_nacimientos.groupby("Año")["Total"].sum()
nacimientos

Los nacimientos de  2020  son:  1625527
Los nacimientos de  2021  son:  1903393
Los nacimientos de  2022  son:  1880718


Año
2020    1625527
2021    1903393
2022    1880718
Name: Total, dtype: int64

### ¿Cuántos nacimientos hubo en total en 2020, 2021 y 2022?

- Los nacimientos de  2020  son:  1625527
- Los nacimientos de  2021  son:  1903393
- Los nacimientos de  2022  son:  1880718
- En total son: 5409638 

In [35]:
df_nacimientos["Total"].sum()

np.int64(5409638)

### ¿En qué año se registró el mayor número de nacimientos?
Los nacimientos de 2021 son: 1903393 corresponden al mayor número de nacimientos.

### ¿En qué año se registró el menor número de nacimientos?
Los nacimientos de 2020 son: 1625527 corresponden al menor número de nacimientos.

### ¿La cantidad de nacimientos parece aumentar o disminuir con el tiempo?
Parece tener tendencia creciente. Se tendría que hacer un análisis de regresión.

## Tarea 2. Tasa de nacimientos por estado
nacimientos_2020.csv y el archivo poblacion.csv, realiza un análisis de la tasa de nacimientos por cada 100,000 habitantes del año 2020. Realiza lo siguiente:
1. Lee la base de población.
2. Une la base de nacimientos con la base de población usando la columna del estado.
3. Crea una nueva columna llamada "Tasa_nacimientos" definida como: Tasa_nacimientos = Nacimientos / Poblacion * 1_000
4. Calcula la tasa de nacimientos por estado y año.
5. Identifica los 5 estados con mayor tasa de nacimientos en 2022.
6. Identifica los 5 estados con menor tasa de nacimientos en 2022.
7. Grafica la tasa de nacimientos por estado para 2022.
Preguntas para responder:
- ¿Qué estado tuvo la mayor tasa de nacimientos en 2022?
- ¿Qué estado tuvo la menor tasa de nacimientos en 2022?
- ¿Los estados con más nacimientos absolutos son necesariamente los estados con mayor tasa?
- ¿Por qué es útil dividir entre la población?